data-prepper-plugins/elasticsearch/src/main/java/com/amazon/dataprepper/plugins/sink/elasticsearch/ConnectionConfiguration.java

```diff
@@ -338,6 +339,8 @@ public Builder withAwsRegion(final String awsRegion) {
     }
 
     public Builder withAWSStsRoleArn(final String awsStsRoleArn) {
+      checkArgument(awsStsRoleArn == null || awsStsRoleArn.length() <= 2048, "awsStsRoleArn length cannot exceed 2048");
+      checkArgument(awsStsRoleArn == null || awsStsRoleArn.equals(Arn.fromString(awsStsRoleArn).toString()), "Invalid ARN format for awsStsRoleArn");
```

Comment:
Agree. We should also replace L343 with try-catch to throw the desired message "Invalid ARN format for awsStsRoleArn". But no need for null check since L342 did the work.

Reply:
@sshivanii I think the idea is to throw the more indicative message "Invalid ARN format for awsStsRoleArn" instead of throwing the side effect message from Arn.fromString